### PR TITLE
update workflows

### PR DESCRIPTION
**Pull request by sean-m-sullivan**
# What does this PR do?
update workflows

### CODE REVIEW

**Comment by sean-m-sullivan** [comment]
@Tompage1994 I am going to need help on this one.

**Comment by Tompage1994** [comment]
Love this. So much testing going on. Works especially well here with the modules. 

I think I've fixed all the Python stuff now and pushed to your branch. Should be good 🤞

**Comment by Tompage1994** [comment]
Ok, so issues lies with the fact the collection is built for the sanity check as infra.ah_configuration but the doc_fragment is set as redhat_cop.ah_configuration.

This is symptomatic of a problem we have with releasing modules here. The doc_fragments only work if the name of the collection is hardcoded in. We need to make sure the the below line in ALL the modules and plugins is replaced with infra for the infra.ah_configuration collection.

```
extends_documentation_fragment: redhat-cop.ah_configuration.auth
```

**Comment by Tompage1994** [comment]
Created #175 for this issue

**Comment by Tompage1994** [comment]
Ok, I've updated all references to redhat_cop.ah_configuration to infra.ah_configuration to make this work.

I don't know whether we want to go with that though. It will mean that we have to fix the doc_fragments for releases to redhat_cop namespace

**Comment by Tompage1994** [comment]
I also removed Black as the Python linter since the ansible-test uses pep8 and the action didn't like ignoring it.

**Comment by Tompage1994** [comment]
ok, lots of action but this should be good now. To summarise the key things that actually change here:

- changed all references of `redhat_cop.ah_configuration` to `infra.ah_configuration`
- Removed Black as a linter

I've approved this PR, but now given I've done a lot of changes too we probably need a third pair of eyes

**Review by Tompage1994 — APPROVED** [review]

**Comment by djdanielsson** [comment]
The workflow can sed the file to fix it for the 6 months we support both namespaces.... yes it's a hack but it should work

**Comment by sean-m-sullivan** [comment]
We can add that to the workflow yes. Was going to suggest we change the variable, but Tom's made it work. I want to run tests on things too, just in case any of the lint changes broke something in the python.

File: changelogs/fragments/linting.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+bugfixes:
+  - Fixed doc issue in ah_collection role where overwrite_existing option was reported with default True when functionality was False
+  - Fixed doc issue in ah_collection_upload role where wait option was reported with default True when functionality was False
+...
```
Comment (djdanielsson) [inline changelogs/fragments/linting.yml:13]: also fixed issue where some private things were not set to no_log, that's fairly big and worth noting.

**Review by djdanielsson — CHANGES_REQUESTED** [review]
this seems like a really big update, though I know a lot of it is just renaming or changing format. I didn't notice anything jump out at me except for changelog not talking about adding no_log to a few vars which I think is valuable.

File: changelogs/fragments/linting.yml (hunk 1 above)
Comment (Tompage1994) [reply changelogs/fragments/linting.yml:13]: Your right, I forgot to continue adding these changelogs for all the things I fixed. I'll add those now

**Comment by Tompage1994** [comment]
Updated with changelogs and a couple of fixes to docs in roles and a minor change to the role behaviour which didnt match the module

**Review by djdanielsson — APPROVED** [review]
LGTM

**Review by sean-m-sullivan — COMMENTED** [review]
LGTM, have given it a few passes over the morning.

**Comment by Tompage1994** [comment]
Opened #176 for outstanding CI failure

**Pull request merged**